src/main/java/frc/robot/networktables/DeltaDebugBoard.java

```diff
@@ -0,0 +1,76 @@
+// Temporary path & name for the file.
+package frc.robot.networktables;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Robot;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.function.Consumer;
+
+public class DeltaDebugBoard<T> {
+  private static Hashtable<String, String> stringTable = new Hashtable<>();
+  private static Hashtable<String, Double> doubleTable = new Hashtable<>();
+  private static Hashtable<String, Boolean> booleanTable = new Hashtable<>();
+
+  private static final DataLog datalog = DataLogManager.getLog();
+  private static final HashMap<String, GenericEntry> entries = new HashMap<String, GenericEntry>();
+
+  private ShuffleboardTab networkTab;
+  private GenericEntry networkEntry;
+  private Integer localEntry;
+  private Consumer<T> localConsumer;
+  private final String name;
+  private T lastValue;
+
+  public DeltaDebugBoard(T defaultValue, String name, SubsystemBase subsystem) {
+    this.name = name;
+    this.lastValue = defaultValue;
+    if (defaultValue instanceof Double) {
```

Comment:
Would be useful to catch and report errors on attempts to use an unsupported data type.